core/vote/vote_manager.go

```diff
@@ -132,11 +135,22 @@ func (voteManager *VoteManager) loop() {
 
 			curHead := cHead.Block.Header()
 			// Check if cur validator is within the validatorSet at curHead
-			if !voteManager.engine.IsActiveValidatorAt(voteManager.chain, curHead) {
+			if !voteManager.engine.IsActiveValidatorAt(voteManager.chain, curHead,
+				func(bLSPublicKey *types.BLSPublicKey) bool {
```

Comment:
Question: Why use a custom function here when all you need to check is if the `blsPublicKey` matches `voterManager.signer.Pubkey` ? Are there any other use cases?

Reply:
> Question: Why use a custom function here when all you need to check is if the `blsPublicKey` matches `voterManager.signer.Pubkey` ? Are there any other use cases?

as before, `IsActiveValidatorAt` only check consensus key 
after fast finality feature brought in,  IsActiveValidatorAt may want to check votekey,
so a new param `checkVoteKeyFn` is added, 
      if it's nil, meaning no need to check; if it's not nil, then should check
if in instead, we pass a pointer `* blsPublicKey`, when we pass nil, the meaning is confusing:
    `nil` meaning not to checking or desire votoKey is nil?

Reply:
I think you can define `isActiveValidatorAt(chain consensus.ChainHeaderReader, header *types.Header, blsPublicKey *types.BLSPublicKey) bool` , and pass `voteManager.signer.Pubkey[:]`  to it.  The implementation of `isActiveValidatorAt` should check that this voting key matches the validator's vote address.  

This way you don't need the custom function.